### PR TITLE
Test api cred

### PR DESCRIPTION
**Pull request by dthampy**
### Problem:
The Dev Console team plans on introducing a change to the createAPICredential API so that it will not return an existing credential. Instead, it will simply fail with an appropriate error if one already exists. This is a will be a breaking change for the API Mesh CLI plugin.

## How Has This Been Tested?
There are two workflows that need to be tested with this change.

If apiKey is available for a particular workspace, then instead of creating a new one, we use that exiting credential and subscribe that key to the meshService. 
This can be tested by creating a new mesh on an exising workspace

If no apiKey is found for a particular workspace, in that case, we would create a new credential and use that and subscribe to the API mesh service
This workflow can be tested by creating a new workspace in your project and continue with creating a new mesh there

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/commands/api-mesh/__tests__/create.test.js [hunk 1]
```diff
@@ -288,27 +294,24 @@ describe('create command tests', () => {
 		  },
 		]
 	`);
-		expect(createAPIMeshCredentials.mock.calls[0]).toMatchInlineSnapshot(`
+		/*expect(createAPIMeshCredentials.mock.calls[0]).toMatchInlineSnapshot(`
 		[
 		  "1234",
 		  "5678",
 		  "123456789",
 		]
```
Comment (amolina-adobe) [inline src/commands/api-mesh/__tests__/create.test.js:302]: Can we just delete this?

File: src/commands/api-mesh/__tests__/create.test.js [hunk 2]
```diff
@@ -288,27 +294,24 @@ describe('create command tests', () => {
...
+	`);*/
+		/*expect(subscribeCredentialToMeshService.mock.calls[0]).toMatchInlineSnapshot(`
 		[
 		  "1234",
 		  "5678",
 		  "123456789",
 		  "dummy_id",
 		]
-	`);
+	`);*/
```
Comment (amolina-adobe) [inline src/commands/api-mesh/__tests__/create.test.js:311]: This too.

**Review by amolina-adobe — CHANGES_REQUESTED** [review]

File: src/lib/devConsole.js [hunk 3]
```diff
@@ -212,7 +212,64 @@ const createMesh = async (organizationId, projectId, workspaceId, data) => {
 
 		if (response && response.status === 201) {
 			logger.info(`Mesh Config : ${objToString(response, ['data'])}`);
-			return response.data;
+
+			let sdkList = [];
+			let credential;
+
+			credential = await getApiKeyCredential(organizationId, projectId, workspaceId);
+			if (credential) {
```
[the remark this answers is not in the document]
Comment (dthampy) [reply src/lib/devConsole.js]: Done

File: src/lib/devConsole.js [hunk 4]
```diff
@@ -212,7 +212,64 @@ const createMesh = async (organizationId, projectId, workspaceId, data) => {
...
+				} else {
+					logger.error('Unable to subscribe API Key %s to API Mesh service', credential.client_id);
+				}
+				return {
+					mesh: response.data,
+					apiKey: credential.client_id,
+					sdkList,
+				};
+			} else {
+				logger.error('API Key credential not found on workspace');
```
[the remark this answers is not in the document]
Comment (dthampy) [reply src/lib/devConsole.js]: Done

File: src/commands/api-mesh/__tests__/create.test.js (hunk 1 above)
Comment (dthampy) [reply src/commands/api-mesh/__tests__/create.test.js]: Done

File: src/commands/api-mesh/__tests__/create.test.js (hunk 2 above)
Comment (dthampy) [reply src/commands/api-mesh/__tests__/create.test.js]: Done

**Review by amolina-adobe — APPROVED** [review]

**Review by jward-magento — APPROVED** [review]
I confirmed that the creation of the new key works when the workspace does not have a pre-existling key.

**Review by brasewel — DISMISSED** [review]

**Review by jward-magento — DISMISSED** [review]

**Review by jward-magento — COMMENTED** [review]
Should this be rev'd all the way to 3.3.3 or should it be like 3.0.1?

**Comment by dthampy** [comment]
> Should this be rev'd all the way to 3.3.3 or should it be like 3.0.1?

Good call @jward-magento
I believe 3.0.0 should be good
@revanth0212 , do you agree?

**Review by amolina-adobe — APPROVED** [review]

**Review by brasewel — APPROVED** [review]

**Review by revanth0212 — APPROVED** [review]

**Pull request merged**